server/jetstream_cluster.go

```diff
@@ -3140,13 +3140,10 @@ func (js *jetStream) applyStreamEntries(mset *stream, ce *CommittedEntry, isReco
 				}
 			}
 
-			if !isRecovering && !mset.IsLeader() {
+			if isRecovering || !mset.IsLeader() {
 				if err := mset.processSnapshot(ss); err != nil {
```

Comment:
This could be dangerous I think since on recover we could think we need to delete legit data..

Reply:
Looked a bit closer, could be ok.